governance-at-scale-account-factory/move-to-ou/v1/src/handler.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import json, logging
+from urllib.request import Request, urlopen
+from betterboto import client as betterboto_client
+import os
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def handler(event, context):
+    request_type = event["RequestType"]
+    try:
+        logger.info(request_type)
+
+        assumable_role_in_root_account_arn = os.environ.get(
+            "ASSUMABLE_ROLE_IN_ROOT_ACCOUNT_ARN"
+        )
+        with betterboto_client.CrossAccountClientContextManager(
+                "organizations", assumable_role_in_root_account_arn, "assumable_org_role"
+        ) as organizations:
+            if request_type in ["Create", "Update"]:
+                target_ou = event.get("ResourceProperties").get("TargetOU")
+                account_id = event.get("ResourceProperties").get("AccountId")
+
+                result = organizations.list_parents_single_page(ChildId=account_id)
+                if len(result.get("Parents", [])) != 1:
+                    raise Exception(
+                        f"There were unexpected parents for the account_id {account_id}: {json.dumps(result)}"
+                    )
+                current_ou = result.get("Parents")[0].get('Id')
+                if str(target_ou) != "None" and current_ou != target_ou:
+                    logger.info("Moving account to new OU")
+                    response = organizations.list_roots()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_roots` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)